File: app/src/main/java/org/breezyweather/background/polling/services/basic/AwakeForegroundUpdateService.java
package org.breezyweather.background.polling.services.basic;

import android.content.Context;

import java.util.List;

import dagger.hilt.android.AndroidEntryPoint;
import org.breezyweather.BreezyWeather;
import org.breezyweather.common.basic.models.Location;
import org.breezyweather.remoteviews.NotificationHelper;
import org.breezyweather.remoteviews.WidgetHelper;
import org.breezyweather.background.polling.PollingManager;

/**
 * Awake foreground update service.
 * */

@AndroidEntryPoint
public class AwakeForegroundUpdateService extends ForegroundUpdateService {

    @Override
    public void updateView(Context context, Location location) {
        WidgetHelper.updateWidgetIfNecessary(context, location);
    }

    @Override
    public void updateView(Context context, List<Location> locationList) {
        WidgetHelper.updateWidgetIfNecessary(context, locationList);
        NotificationHelper.updateNotificationIfNecessary(context, locationList);
    }

    @Override
    public void handlePollingResult(boolean failed) {
        PollingManager.resetAllBackgroundTask(this, false);
    }

    @Override
    public int getForegroundNotificationId() {
        return BreezyWeather.NOTIFICATION_ID_UPDATING_AWAKE;
    }
}

File: app/src/main/java/org/breezyweather/background/polling/services/basic/AwakeForegroundUpdateService.kt
package org.breezyweather.background.polling.services.basic

import android.content.Context
import dagger.hilt.android.AndroidEntryPoint
import org.breezyweather.background.polling.PollingManager.resetAllBackgroundTask
import org.breezyweather.common.basic.models.Location
import org.breezyweather.remoteviews.Notifications
import org.breezyweather.remoteviews.Widgets

/**
 * Awake foreground update service.
 */
@AndroidEntryPoint
class AwakeForegroundUpdateService : ForegroundUpdateService() {
    override fun updateView(context: Context, location: Location) {
        Widgets.updateWidgetIfNecessary(context, location)
    }

    override fun updateView(context: Context, locationList: List<Location>) {
        Widgets.updateWidgetIfNecessary(context, locationList)
        Notifications.updateNotificationIfNecessary(context, locationList)
    }

    override fun handlePollingResult(updateSucceed: Boolean) {
        resetAllBackgroundTask(this, false)
    }

    override val foregroundNotificationId = Notifications.ID_UPDATING_AWAKE
}
